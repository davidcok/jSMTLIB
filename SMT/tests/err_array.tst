; tests bad use of array functions and sort
(set-logic QF_UF)
(declare-sort I 0)
(declare-sort V 0)
(declare-fun a () (Array I V))
(declare-fun a () I)
(declare-fun i () I)
(declare-fun v () V)
(assert (= v (select a i)))
(assert (= a (store a i v)))
