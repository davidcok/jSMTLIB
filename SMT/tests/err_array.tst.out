success
success
success
(declare-fun a () (Array I V))
                   ^^^^^
(error "No such sort symbol declared: Array")
success
success
success
(assert (= v (select a i)))
             ^^^^^^^^^^^^
(error "Unknown predicate symbol select with argument types I I")
(assert (= a (store a i v)))
             ^^^^^^^^^^^^^
(error "Unknown predicate symbol store with argument types I I V")
